pkg/quota/admission/runonceduration/admission.go

```diff
@@ -0,0 +1,131 @@
+		return NewRunOnceDuration(pluginConfig), nil
+	})
+}
+
+func readConfig(reader io.Reader) (*RunOnceDurationConfig, error) {
+	config := &RunOnceDurationConfig{}
+	if reader == nil || reflect.ValueOf(reader).IsNil() {
+		return config, nil
+	}
+	configBytes, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return nil, err
+	}
+	err = configlatest.ReadYAML(configBytes, config)
+	if err != nil {
+		return nil, err
+	}
+	errs := ValidateRunOnceDurationConfig(config)
+	if len(errs) > 0 {
+		return nil, errs.ToAggregate()
+	}
+	return config, nil
+}
+
+func NewRunOnceDuration(config *RunOnceDurationConfig) admission.Interface {
+	return &runOnceDuration{
+		Handler: admission.NewHandler(admission.Create, admission.Update),
+		config:  config,
+	}
+}
+
+type runOnceDuration struct {
+	*admission.Handler
+	config *RunOnceDurationConfig
+	cache  *projectcache.ProjectCache
+}
+
+var _ = oadmission.WantsProjectCache(&runOnceDuration{})
+var _ = oadmission.Validator(&runOnceDuration{})
+
+func (a *runOnceDuration) Admit(attributes admission.Attributes) error {
+	if a.config == nil {
+		return nil
+	}
+	if a.config.ActiveDeadlineSecondsOverride == nil && !a.config.AllowProjectAnnotationOverride {
+		return nil
+	}
+	if attributes.GetResource() != kapi.Resource("pods") {
+		return nil
+	}
+	if len(attributes.GetSubresource()) > 0 {
+		return nil
+	}
+	pod, ok := attributes.GetObject().(*kapi.Pod)
+	if !ok {
+		return admission.NewForbidden(attributes, fmt.Errorf("unexpected object: %#v", attributes.GetObject()))
+	}
+	if pod.Spec.RestartPolicy != kapi.RestartPolicyNever && pod.Spec.RestartPolicy != kapi.RestartPolicyOnFailure {
+		return nil
+	}
+
+	appliedProjectOverride := false
+	if a.config.AllowProjectAnnotationOverride {
+		var err error
+		appliedProjectOverride, err = a.applyProjectAnnotationOverride(attributes.GetNamespace(), pod)
+		if err != nil {
+			return admission.NewForbidden(attributes, err)
+		}
+	}
+
+	if !appliedProjectOverride && a.config.ActiveDeadlineSecondsOverride != nil {
+		pod.Spec.ActiveDeadlineSeconds = a.config.ActiveDeadlineSecondsOverride
+	}
+	return nil
+}
+
+func (a *runOnceDuration) SetProjectCache(cache *projectcache.ProjectCache) {
+	a.cache = cache
+}
+
+func (a *runOnceDuration) Validate() error {
+	if a.cache == nil {
+		return errors.New("RunOnceDuration plugin requires a project cache")
+	}
+	return nil
+}
+
+func (a *runOnceDuration) applyProjectAnnotationOverride(namespace string, pod *kapi.Pod) (bool, error) {
+	ns, err := a.cache.GetNamespace(namespace)
+	if err != nil {
+		return false, fmt.Errorf("error looking up pod namespace: %v", err)
+	}
+	if ns.Annotations == nil {
+		return false, nil
+	}
+	override, hasOverride := ns.Annotations[ActiveDeadlineSecondsOverrideAnnotation]
+	if !hasOverride {
+		return false, nil
+	}
+	overrideInt64, err := strconv.ParseInt(override, 10, 64)
+	if err != nil {
+		return false, fmt.Errorf("cannot parse the ActiveDeadlineSeconds override for project %s: %v", ns.Name, err)
```

Comment:
Minor: consider reporting the value that couldn't be parsed so whomever's looking at the error doesn't need to go look it up.